framework/event_system/Event.h

```diff
@@ -0,0 +1,30 @@
+#ifndef CHITECH_EVENT_H
+#define CHITECH_EVENT_H
+
+#include "parameters/parameter_block.h"
+
+#include <string>
+
+namespace chi
+{
+
+class Event
+{
+public:
+  explicit Event(const std::string& name, int code);
+  explicit Event(const std::string& name, int code,
+                 const ParameterBlock& parameter_block);
```

Comment:
Isn't explicit generally for one argument constructors? 
```suggestion
  Event(const std::string& name, int code);
  Event(const std::string& name, int code,
                 const ParameterBlock& parameter_block);
```

Reply:
No. You can use it to make every argument explicit. But in this case I agree we probably don't need it.